### PR TITLE
mariadb 11.1.2

### PR DESCRIPTION
**Pull request by chenrui333**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by makigumo** [comment]
Just FYI.
The issue seems to be with the recent fmt version 10.1.0.
Building with the previous version 10.0.0 seems to be fine.

**Comment by ZhongRuoyu** [comment]
> The issue seems to be with the recent fmt version 10.1.0.

Yes. The reason is that `fmt::detail::make_arg` [no longer accepts temporaries](https://github.com/fmtlib/fmt/issues/3596#issuecomment-1679244746).

**Comment by ZhongRuoyu** [comment]
To be safe, I suggest waiting for upstream maintainers' feedback on the patch first.

**Comment by chenrui333** [comment]
Nice work!

- relates to https://github.com/MariaDB/server/pull/2732

File: Formula/m/mariadb.rb
Comment (samford) [inline Formula/m/mariadb.rb:4]: downloads.mariadb.com files are now found at dlm.mariadb.com and the new URL for this file is https://dlm.mariadb.com/3399259/MariaDB/mariadb-11.0.3/source/mariadb-11.0.3.tar.gz.

That said, 11.1.2 appears to be the newest stable (GA) version as of 2023-08-21.

Comment (samford) [inline Formula/m/mariadb.rb:4]:
```suggestion
  url "https://archive.mariadb.org/mariadb-11.0.3/source/mariadb-11.0.3.tar.gz"
```

**Review by samford — COMMENTED** [review]
For what it's worth, 11.1.2 appears to be the newest stable version as of yesterday (2023-08-21). [For reference, the archive.mariadb.org tarball URL for 11.1.2 is `https://archive.mariadb.org/mariadb-11.1.2/source/mariadb-11.1.2.tar.gz`].

File: Formula/m/mariadb.rb
Comment (ZhongRuoyu) [reply Formula/m/mariadb.rb:4]: Will do after the fix is finalised. (11.1.2 is also affected by the `libfmt` issue.)

Same for the remaining `mariadb` PRs.

Comment (ZhongRuoyu) [reply Formula/m/mariadb.rb]: Applied suggestion (and updated to 11.1.2).

File: Formula/m/mariadb.rb
Comment (ZhongRuoyu) [inline Formula/m/mariadb.rb:6]: Corrected the license; see https://github.com/MariaDB/server/blob/mariadb-11.1.2/COPYING#L299

Comment (samford) [reply Formula/m/mariadb.rb:6]: @ZhongRuoyu The "or...later" text in the GPL `COPYING` file is provided as an example and isn't an indication of whether the project is `-only` or `-or-later`. Instead, it's necessary to look through other files (e.g., `README`, leading comments in source files, etc.) to see if they use the "or...later" language. Some projects clearly state that the license is only a specific GPL version but others are implicitly "only" through absence of the "or...later" language in license text [outside of the `COPYING` file].

In this case, the `README` contains a "Licensing" section that explicitly states:

> MariaDB is specifically available only under version 2 of the GNU General Public License (GPLv2). (I.e. Without the "any later version" clause.) This is inherited from MySQL. Please see the README file in the MySQL distribution for more information.

With that in mind, the license for the mariadb formulae should be reverted back to `GPL-2.0-only`.

Comment (ZhongRuoyu) [reply Formula/m/mariadb.rb:6]: Sorry for the oversight @samford. Have opened #140689 to correct the other `mariadb` formulae; will update the remaining soon.

**Review by iMichka — APPROVED** [review]

**Pull request merged**
